### PR TITLE
[ROCm] CUDA/ROCm shared interface

### PR DESCRIPTION
**Pull request by ekuznetsov139**
This PR adds several macros and typedefs to assist with writing kernels that seamlessly work with both CUDA and ROCm.

### CODE REVIEW

**Comment by deven-amd** [comment]
gentle ping

**Comment by deven-amd** [comment]
gentle ping

File: tensorflow/core/util/gpu_device_functions.h [hunk 1]
```diff
@@ -33,12 +35,63 @@ limitations under the License.
...
+using gpuFloatComplex = cuFloatComplex;
+using gpuDoubleComplex = cuDoubleComplex;
+#define gpuEventRecord cudaEventRecord
+#define gpuEventSynchronize cudaEventSynchronize
+#define gpuEventDestroy cudaEventDestroy
+#define gpuEventCreate cudaEventCreate
+#define gpuEventCreateWithFlags cudaEventCreateWithFlags
+#define gpuEventDisableTiming cudaEventDisableTiming
+typedef cudaStream_t gpuStream_t;
+typedef cudaEvent_t gpuEvent_t;
```
Comment (cheshire) [inline tensorflow/core/util/gpu_device_functions.h]: Could those be done using `using` for consistency?

**Review by cheshire — DISMISSED** [review]

**Comment by gbaned** [comment]
@ekuznetsov139 Can you please check @cheshire's comments and keep us posted. Thanks!

**Review by cheshire — APPROVED** [review]

**Pull request merged**